### PR TITLE
Drop the dependency on `@jupyterlab/buildutils` in `@jupyterlab/builder`

### PR DESCRIPTION
**Pull request by jtpio**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/13740

This will help reduce the number of security vulnerabilities in downstream extensions and applications.

## Code changes

Drop dependency on `@jupyterlab/buildutils` in `@jupyterlab/builder`.

Replaces the uses of `readJSONFile` and `writeJSONFile` by the equivalent from `fs-extra`.

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

File: builder/src/extensionConfig.ts [hunk 1]
```diff
@@ -203,7 +202,7 @@ function generateConfig({
           _build.style = './style';
         }
         data.jupyterlab._build = _build;
-        writeJSONFile(path.join(outputPath, 'package.json'), data);
+        fs.writeJSONSync(path.join(outputPath, 'package.json'), data);
```
[the remark this answers is not in the document]
Comment (jtpio) [reply builder/src/extensionConfig.ts]: Although that part of the builder logic corresponds to the copy of the `package.json` to the `outputDir` folder of the lab extension with the module federation metadata. Maybe that `package.json` does not need to be sorted.

[the remark this answers is not in the document]
Comment (jtpio) [reply builder/src/extensionConfig.ts]: Looking at the git history, this builder logic was added in https://github.com/jupyterlab/jupyterlab/pull/9040, but doesn't say anything about the file having to be sorted.

**Comment by jtpio** [comment]
Marking as ready for review.

I would say it should be fine to drop the sort logic mentioned in the comment above (https://github.com/jupyterlab/jupyterlab/pull/13741#pullrequestreview-1240850663) to keep things simple and avoid adding other dependencies or duplicating existing code.

**Review by fcollonval — APPROVED** [review]
Thanks @jtpio

**Comment by fcollonval** [comment]
Should this be backported to 3.x

**Comment by jtpio** [comment]
> Should this be backported to 3.x

Normally it should be fine yes. Maybe the only potential issue could be extensions relying on a transitive dependency pulled via `@jupyterlab/buildutils` without having it properly defined in `dependencies`.

**Comment by bollwyvl** [comment]
> relying on a transitive dependency 

I'm going to wager the number of projects using it is very small, and any issue will be caught at build/test time, so has no real downstream user impact. 

> Should this be backported to 3.x

_please_ backport to 3.5.x.

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.6.x

**Comment by fcollonval** [comment]
@bollwyvl I'll will backport it to 3.6.x and then on 3.5.x as the bot is failing to do it automatically.

**Pull request merged**